draft-ietf-httpbis-http2-secondary-certs.md

```diff
@@ -775,6 +803,13 @@ its control in order to present the compromised certificate. As recommended in
 {{?RFC8336}}, clients opting not to consult DNS ought to employ some alternative
 means to increase confidence that the certificate is legitimate.
 
+One such means is the Required Domain certificate extension defined in
+{extension}. Clients MUST require that server certificates presented via this
+mechanism contain the Required Domain extension and require that a certificate
+previously accepted on the connection (including the certificate presented in
+TLS) lists the Required Domain in the Subject field or the Subject Alternative
+Name extension.
```

Comment:
I think that you want to explicitly address the question ekr raised at the meeting (which might not have been minuted).

He asks whether you can produce an EA that has a certificate without a required domain and then use that as the target of required domain in other certificates.  I don't think that this is a good idea because it allows an attacker to muddy the audit trail if they are able to acquire a certificate they control. But we should be very clear either way and this isn't quite clear enough.

"Any certificate presented in an exported authenticator MUST be rejected if it does not include a RequiredDomain extension." might be enough.

Reply:
Except that's not quite correct, because we're not requiring this in the client->server direction.  I thought ekr's concern was already covered by this requirement:

> Clients MUST require that server certificates presented via this
mechanism contain the Required Domain extension and require that a certificate
previously accepted on the connection (including the certificate presented in
TLS) lists the Required Domain in the Subject field or the Subject Alternative
Name extension.

That is, a certificate with no Required Domain will never be an acceptable secondary to clients and therefore can't have been previously accepted (except in the TLS handshake), but can still be acceptable to servers for client verification.

Reply:
I added a reference to this requirement in the section about validating the contents of an EA.  Hopefully that addresses the concern.